pkg/tsdb/azuremonitor/loganalytics/azure-log-analytics-datasource.go

```diff
@@ -142,12 +141,12 @@ func (e *AzureLogAnalyticsDatasource) ExecuteTimeSeriesQuery(ctx context.Context
 	for _, query := range originalQueries {
 		logsQuery, err := e.buildQuery(ctx, query, dsInfo, fromAlert)
 		if err != nil {
-			errorsource.AddErrorToResponse(query.RefID, result, err)
+			result.Responses[query.RefID] = backend.ErrorResponseWithErrorSource(err)
```

Comment:
When I see this in action now, I do think it assumes too much - such as people knowing what "error source" is and also that any error you pass here should be an error with a source - which forces you to dig through nested calls to figure out if we will always return that type or not and may end up with people having code like this everywhere https://github.com/grafana/grafana/pull/97163/files#diff-4a1a258d4df7b924594e883db0618f2d811400c0b9161a722a3f5749397481bcR58 🥴

Reply:
> When I see this in action now, I do think it assumes too much - such as people knowing what "error source" 

Do I understand it correctly that you are referring to the naming of the function or how it works?

Reply:
At first it was just the name, but I think it's a combination 😅 It's just my feeling on it, but may be okay in practice. In reality it's just a pattern people will have to get used to - not really sure if there's any better way to make this any more idiomatic.

Reply:
Yeah, it is tricky, because I think if the name would be just something like `backend.ErrorResponse(err)` and it would be adding error source under the hood, some developers could be adding ErrorSource manually. But the this name requires for you to know about error source. But also, you kind of need to know about error source at this point 😅. And if you don't, there are other methods that you can use.

Also this method does not require error to have source. If it does not, then it just not add it.

Reply:
Yeah I agree it's a tricky balance - but we have to start somewhere I guess 😄 Feel free to add an alias for this func to the SDK with the updated name. Otherwise if you feel like it's probably okay then 🚀